### PR TITLE
feat(configcat-provider): Add ConfigCat provider

### PR DESCRIPTION
**Pull request by rcrowe**
## This PR

- Adds [ConfigCat](https://configcat.com/) provider

### Related Issues

N/A

### Notes

Community contribution working with @z4kn4fein the author of the ConfigCat Go SDK.

### Follow-up Tasks

N/A

### How to test

- Standard Go tests
- `make test` from repo root

### CODE REVIEW

File: providers/configcat/go.mod
Comment (Kavindu-Dodan) [inline providers/configcat/go.mod:3]: Minor - I hope this is to be compatible with configcat go-sdk v7 and to support old platforms?

**Review by Kavindu-Dodan — APPROVED** [review]
LGTM 🚀 

Provider implementation as well as tests are nicely done. Thank you for the contribution for OpenFeature 🤝

File: providers/configcat/go.mod
Comment (rcrowe) [reply providers/configcat/go.mod:3]: That's correct I noticed [a comment from you](https://github.com/open-feature/go-sdk-contrib/pull/128#discussion_r1128585782) that seemed to suggest it was best to align with the underlying SDK.

Should this instead align with [go1.18](https://github.com/open-feature/go-sdk/blob/d93f58bfb9ea92fbcbb0faa7cd7880bb629a6ebe/go.mod#L3) from the `go-sdk`?

**Comment by rcrowe** [comment]
Thank you @Kavindu-Dodan for your help reviewing this.

Do you know what release schedule this project works too, so we can get this merged & a tagged release?

**Comment by Kavindu-Dodan** [comment]
> Thank you @Kavindu-Dodan for your help reviewing this.
> 
> Do you know what release schedule this project works to, so we can get this merged & a tagged release?

Welcome. I am waiting for at least one more review and then we can move ahead and merge this.

And as soon as we merge this, release please will create a release PR (for example - [1])

[1] - https://github.com/open-feature/go-sdk-contrib/pull/216

**Pull request merged**
